add documention on grafana backups (#2558)
giantswarm/docs: src/content/tutorials/observability/data-exploration/grafana-backups/_index.md

---

> linkTitle: Grafana backups

Reviewer: I think this belongs to the intranet right?

Reviewer: I don't see a reason why customers would create a backup

Author: I thought they might want to have some control over it :shrug:

Reviewer 2: It 100% makes sense to me that customers might want to tweak things or back up on demand, BUT

If backups are to be customer configurable, I'd strongly prefer if there's an o11y platform way to configure it so that customers don't need to touch CNPG primitives. We aren't providing CNPG as a service to them, it should just be an internal implementation detail.

Author: Ok makes sense to me :+1: 

I'll just keep the change concerning the dashboard creation method then and remove this for now.

Reviewer 2: Thanks :pray: If you want, the backup reference material could still go to internal docs, for other folks working with CNPG

Author: yeah was planning on it :)

title: Creating and using grafana backups
description: Guide explaining how to create and use grafana backups in the Observability Platform for disaster recovery.
menu:
  principal:
    identifier: tutorials-observability-data-exploration-grafana-backups
    parent: tutorials-observability-data-exploration
weight: 40
last_review_date: 2025-04-07
user_questions:
  - How to create grafana backups?
  - How to recover grafana data after a disater?
owner:
  - https://github.com/orgs/giantswarm/teams/team-atlas
---

> We are operating a postgresql cluster as a grafana database on which is stored all grafana data. In order to ensure the availability of this stored data, especially in case of disaster, backups need to be created so that the data can be retrieved afterwards. We are already creating backups by default but if you want to add your own, you will find here how to do it.

Reviewer 3: ```suggestion
We operate a PostgreSQL cluster as a Grafana database storing your data. To ensure the availability of this stored data, especially in a disaster, backups need to be created to retrieve the data afterward. We are already creating backups by default, but if you want to add your own, you will find here how to do it.
```

## Creating grafana backups

There are 2 ways to create backups for the grafana postgresql database :

> - __Scheduled backups__ : you can define a `ScheduledBackup` resource in the k8S cluster in which you will define when will backup be created. This is the preferred approach as it allows you to automate backups and have those done on a regular basis. Here is an example of such a resource :

Reviewer 3: ```suggestion
- __Scheduled backups__ : You can define a `ScheduledBackup` resource in the management cluster. This is the preferred approach as it allows you to automate backups and have those done regularly. Here is an example of such a resource :
```

```yaml
apiVersion: postgresql.cnpg.io/v1
kind: ScheduledBackup
metadata:
  name: scheduled-backup-example
spec:
  schedule: "0 0 0 * * *"
  backupOwnerReference: self
  cluster:
    name: my-pg-cluster
```

> The `schedule` field is a cron schedule specification so you can define it the same way you would for a cronjob.
> The `cluster.name` field designates the postgresql cluster you want to create a backup from.

Reviewer 3: ```suggestion
The `schedule` field is a cron schedule specification, so you can define it the same way you would for a cronjob. The `cluster.name` field designates the PostgreSQL cluster from which you want to create a backup.
```

> - __Manual backups__ : if for any reason you need to create a backup outside of the scheduled time from the ScheduledBackups resources, you can create a `Backup` resource. This will create a backup from the targeted postgresql cluster as soon as it gets deployed and reconciled by the postgresql operator. here is an example of such a resource :

Reviewer 3: ```suggestion
- __Manual backups__ : If you need to create a backup outside of the scheduled time from the `ScheduledBackup` resources, you can create a `Backup` resource. This will create a backup from the targeted PostgreSQL cluster as soon as it gets deployed and reconciled by the PostgreSQL operator. Check the example of such a resource :
```
